### ISSUE TITLE
cmake: cmake subproject does not configure with `ERROR: Arguments must be strings.`

### ISSUE THREAD

**Issue by rubensfig** [mesonbuild/meson#8893]
**Describe the bug**
When running meson build with a cmake subproject, the command fails with 

```
Element not a string: [None]

meson.build:14:0: ERROR: Arguments must be strings.
```

Similar behaviour to https://github.com/mesonbuild/meson/issues/7715

**To Reproduce**
Relevant section of the meson.build file:

```
# telnet client
cmake = import('cmake')
telnetclient_proj = cmake.subproject('SimpleTelnetClient')
telnetclient_dep = telnetclient_proj.dependency('SimpleTelnetClient')
```

**Expected behavior**
The project configures everything correctly.

**system parameters**
Fedora 33
meson --version: 0.57.1

**Comment by mensinda** [comment]
Can you please post your `meson-log.txt`, or even better, a simple project for reproducing this problem?

**Comment by mensinda** [comment]
The problem is that you are trying to access an executable with `dependency()`. You have to use target instead. Fixed the error message in #8966.
